.pre-commit-config.yaml

```
@@ -0,0 +1,12 @@
+repos:
+  - repo: https://github.com/pre-commit/mirrors-eslint
```

And once you've disabled all style checks for eslint, which is unhelpful, leave style to be handled by prettier, which does a better job in that regard:

```suggestion
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.1.2
    hooks:
      - id: prettier
  - repo: https://github.com/pre-commit/mirrors-eslint
```

Some important points:

1. I don't think Odoo would like this for v12.
1. Different pre-commit pinned versions across odoo branches would difficult the bw/fw ports.
1. It'd be great if Odoo includes a few more checks or formatters. For python there's https://github.com/akaihola/darker if they don't want to fully black the code.
1. Prettier has a XML plugin we use at OCA that works pretty well too. See https://github.com/OCA/oca-addons-repo-template/pull/41/files#diff-0f76a5aa80b9c2ce9f078384ac3e4bccae3defa883c7183632932e624765de51R70-R79

---

1. A solution could be marking check as failed only when lines changed by pre-commit are intersect with lines changed in pull request.

2. Good point. It means that on updating pre-commit config it has to be done in all maintained branches